src/jdk.incubator.code/share/classes/jdk/incubator/code/bytecode/BytecodeGenerator.java

```diff
@@ -791,6 +794,25 @@ private void generate() {
                         push(op.result());
                     }
                     case InvokeOp op -> {
+                        // Resolve referenced class to determine if interface
+                        MethodRef md = op.invokeDescriptor();
+                        JavaType refType = (JavaType)md.refType();
+                        ClassDesc specialCaller = lookup.lookupClass().describeConstable().get();
+                        MethodTypeDesc mDesc = MethodRef.toNominalDescriptor(md.type());
+                        if (op.invokeKind() == InvokeOp.InvokeKind.SUPER) {
+                            // constructs method handle via lookup.findSpecial using the lookup's class as the specialCaller
+                            // original lookup is stored in class data
+                            // @@@ performance can be improved by storing a list of the resolved method handles instead
+                            cob.ldc(DynamicConstantDesc.of(BSM_CLASS_DATA))
+                               .checkcast(CD_MethodHandles_Lookup)
+                               .ldc(refType.toNominalDescriptor())
+                               .ldc(md.name())
+                               .ldc(mDesc)
+                               .ldc(specialCaller)
+                               .invokevirtual(CD_MethodHandles_Lookup,
+                                              "findSpecial",
+                                              MethodTypeDesc.of(CD_MethodHandle, CD_Class, CD_String, CD_MethodType, CD_Class));
```

Comment:
Performance can be improved (if necessary) by indexing of all resolved method handles and storing them in class data as a list.
Invoke code would not perform search for the method handle, but it will receive it directly from class data.